torch/csrc/distributed/c10d/FlightRecorder.cpp

```diff
@@ -42,10 +42,20 @@ void DebugInfoWriter::write(const std::string& trace) {
 
 DebugInfoWriter& DebugInfoWriter::getWriter(int rank) {
   if (writer_ == nullptr) {
-    // Attempt to write to running user's HOME directory cache folder - if it
-    // exists.
-    auto homeDir = getCvarString({"HOME"}, "/tmp");
-    auto cacheDirPath = c10::filesystem::path(homeDir + "/.cache/torch");
+// Attempt to write to running user's HOME directory cache folder - if it
+// exists.
+#ifdef _WIN32
+    const char* cacheHome = nullptr;
+#else
+    // Uses XDG_CACHE_HOME if it's set
+    const char* cacheHome = std::getenv("XDG_CACHE_HOME");
+#endif
+    std::string cacheRoot;
+    if (cacheHome)
+      cacheRoot = cacheHome;
```

Comment:
braces please

Reply:
Done. Wasn't sure if they should be used for single line blocks.